ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -656,33 +656,205 @@ let text_document_lens (state : State.t)
 
 let folding_range (state : State.t)
```

Comment:
I feel like this is a too involved function (too large at least) and deserves to be moved to a separate file :-)